lecture_1/homework/app/controllers/samurais_controller.rb

```diff
@@ -0,0 +1,67 @@
+class SamuraisController < ApplicationController
+  #before_action :set_clan
+  #before_action :samurai
+
+  def index
+    samurais = clan.samurais.all
+    render json: samurais.to_json(only: %i[id name armour battles])
+  end
+
+  def show
+    render json: samurai.to_json()
+  end
+
+  def create
+    #clan ||= Clan.find(params[:clan_id])
+    samurai = clan.samurais.new(samurai_params)
+    if samurai.save
+      render json: samurai.to_json(only: %i[:id name armour]),
+             status: 201
+    else
+      render json: {errors: samurai.errors },
+             status: 422
+    end
+  end
+
+  def update
+    if samurai.update!(samurai_params)
+      render json: samurai.to_json(only: %i[id name armour]),
+             status: 201
+    else
+      render json: {errors: samurai.errors},
+             status: 422
+    end
+  end
+
+  def alive
```

Comment:
Najlepiej nie tworzyć takich `nierestowych` akcji, tylko obsługiwać je jakoś w odpowiedniej restowej akcji(dostępne w CRUD to show, index, create, update, destroy) albo wydzielić do nowego zagnieżdżonego kontrolera, w naszym przypadku 100% full super restowo ścieżki by wyglądały tak:

```ruby
resources :clans, param: :name  do
  resources :samurais do
    resources :alive, only: %i[index]
    resources :dead, only: %i[index]
  end
end
```

Oczywiście staramy się unikać zagnieżdżeń więc to póki co samo obsługiwanie 2 query params w indexie by wystarczyło, ale jeśli mielibyśmy np. inne akcje `update` dla żywych i martwych samurai, to taką architekturę dużo łatwiej się rozbudowuje i ładniej dzieli tą logikę :D